addons/purchase/migrations/9.0.1.2/post-migration.py

```diff
@@ -194,30 +196,32 @@ def set_po_line_computed_rest(env):
     # be called in end-migration, where mrp module will be available for
     # sure in the environment if it is installed
     openupgrade.logged_query(
-        env.cr,
-        """
-            UPDATE purchase_order_line pol
-            SET qty_received = 0.0
-            FROM purchase_order po
-            WHERE
-                po.id = pol.order_id AND
-                po.state NOT IN ('purchase', 'done')
-        """
-    )
-    openupgrade.logged_query(
-        env.cr,
-        """
-            UPDATE purchase_order_line pol
-            SET qty_received = pol.product_qty
+        env.cr, """
+        UPDATE purchase_order_line pol
+        SET qty_received = sub.qty_received
+        FROM (
+            SELECT CASE
+                WHEN MIN(po.state) NOT IN ('purchase', 'done') THEN 0.0
```

Comment:
what means MIN(po.state)?

Reply:
It's because we are grouping at other level, so we will the same `po.state` across several potential records, so we need this for making it work, but no other consequences.

Reply:
ok